src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,42 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  todoId: number;
+  todos: Todo[];
+  selectedTodo: (value: number) => number | void;
+};
+
+// eslint-disable-next-line max-len
+export const TodoModal: React.FC<Props> = ({
+  todoId,
+  todos,
+  selectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isValidCard, setIsValidCard] = useState(true);
+
+  const currentTodo = todos.find(({ id }) => id === todoId);
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(userFromServer => setUser(userFromServer));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code